tchannel-core/src/main/java/com/uber/tchannel/channels/Peer.java

```diff
@@ -152,31 +150,25 @@ public Connection getConnection(ChannelId channelId) {
     }
 
     public void close() {
-        for (ChannelId id : connections.keySet()) {
-            Connection conn = connections.get(id);
-            if (conn != null) {
-                conn.close();
-            }
+        for ( Connection conn : connections.values()) {
+            conn.close();
         }
-
         this.connections.clear();
+
     }
 
     public Map<String, Integer> getStats() {
         int in = 0;
         int out = 0;
-        for (ChannelId id : connections.keySet()) {
-            Connection conn = connections.get(id);
-            if (conn == null) {
-                continue;
-            } else if (conn.direction == Connection.Direction.OUT) {
+        for (Connection conn : connections.values()) {
+            if (conn.direction == Connection.Direction.OUT) {
                 out++;
             } else {
                 in++;
             }
         }
 
-        Map<String, Integer> result = new HashMap<>();
+        Map<String, Integer> result = new HashMap<>(2);
```

Comment:
You are right about the 2 and resize but the reason is different: while 2 is enough, there is an additional internal HashMap call to resize when the capacity has been reached. I am making it 3. The load factor is not a problem.